extension/intents/navigation/navigation.js

```diff
@@ -13,28 +13,36 @@ intentRunner.registerIntent({
   name: "navigation.navigate",
   async run(context) {
     const query = context.slots.query;
+    const where = context.slots.where;
     const cached = queryDatabase.get(query.toLowerCase());
-    if (cached) {
-      await context.openOrFocusTab(cached.url);
-    } else {
-      const tab = await context.createTabGoogleLucky(query);
-      const url = tab.url;
-      queryDatabase.set(query.toLowerCase(), {
-        url,
-        date: Date.now(),
-      });
-      // Sometimes there's a very quick redirect
-      setTimeout(async () => {
-        const newTab = await browser.tabs.get(tab.id);
-        if (newTab.url !== url) {
+    if (where === "window") {
+       if (cached) {
+        await browser.windows.create({url: cached.url});
+      } else {
+        await browser.windows.create({});
+        await context.createTabGoogleLucky(query);
+      }
+    } else if (cached) {
+        await context.openOrFocusTab(cached.url);
+      } else {
+          const tab = await context.createTabGoogleLucky(query);
+          const url = tab.url;
           queryDatabase.set(query.toLowerCase(), {
-            url: newTab.url,
+            url,
             date: Date.now(),
           });
+          // Sometimes there's a very quick redirect
+          setTimeout(async () => {
+            const newTab = await browser.tabs.get(tab.id);
+            if (newTab.url !== url) {
+              queryDatabase.set(query.toLowerCase(), {
+                url: newTab.url,
+                date: Date.now(),
+              });
+            }
+          }, 1000);
```

Comment:
All of this code should really be run if `where` is a tab or window. So really while there's that `browser.windows.create({})` code that's conditional on `where`, the rest can and should be run regardless.

Reply:
Actually this is creating problem as **tab.id** and **url** will not be defined if put outside the else as the scope is limited within else block. Another way to resolve this is to add a setTimeOut fn within the where===window conditional also . But it is rewriting of code . What you do this I should do?